spring-restdocs-core/src/main/java/org/springframework/restdocs/cli/CurlRequestSnippet.java

```diff
@@ -45,6 +44,8 @@
  */
 public class CurlRequestSnippet extends TemplatedSnippet {
 
+	private CommandFormatter commandFormatter = ConcatenatingCommandFormatter.newlineSeparator();
+
 	/**
 	 * Creates a new {@code CurlRequestSnippet} with no additional attributes.
 	 */
```

Comment:
I was torn on this earlier when I was reviewing the change but, while annoying, I think it's probably the right thing to do. It's what I was getting at when I said "but creating the snippet with a single-line formatter" in my comment on the tests below.

My motivation was that I wanted the opinion on what the default formatter should be to be in a single place. `CliDocumentation` seemed to make sense as it can then pass the default formatter into `CurlRequestSnippet` and `HttpieRequestSnippet`. 

I was also envisaging a further change where the default moves to `SnippetConfigurer` and it'll apply it to the default `CurlRequestSnippet` and `HttpieRequestSnippet` that it creates. The would lead to the deprecation of `curlRequest()` and `httpieRequest()` on `CliDocumentation` as well.

As I said, I was torn on this so if you don't think the reasoning above is sound I'd really like to hear any counter arguments.